### ISSUE TITLE
Add "local only" mode for easier setup

### ISSUE THREAD

**Issue by Grishnackh** [syncthing/syncthing#9377]
Currently discovery and relay are working in a global mode by default. When setting up SyncThing on my NAS and devices (desktop and mobile) I intended to build a sync mechanism that works on my local network only - without any need of a cloud or other kinds of public access.

It was quiet surprising to see SyncThing doing a lot of calls to the public internet. The docs state that they are neccessary for discovery and relay. It seems to be possible to run SyncThing in a "local only" mode, when you configure some parameters the right way which is not very intuitive. For example there are several discussions around the question in different forum threads:

* https://forum.syncthing.net/t/how-can-i-make-sure-syncthing-is-using-local-network-only/20520
* https://forum.syncthing.net/t/local-sync-only/11200
* https://forum.syncthing.net/t/configuration-to-restrict-syncing-only-when-on-same-network/17269
* https://forum.syncthing.net/t/limit-syncthing-to-only-use-local-network-review-and-suggestions-appreciated/20117
* https://blog.b-nm.at/2019/02/27/syncthing-force-local-sync/

It would be really helpful if the setup of SyncThing provides a "local only" mode which configures all the needed settings out of the box. The feature seems to be useful for a braoder audience, especially for less experienced users.

**Comment by calmh** [comment]
Yeah, I was with you all the way until the "especially for less experienced users" who I think are the least suitable target audience given that the features in question are implemented specifically to make things mostly "just work" for them. Nonetheless, sure, a `--stealth-mode` or whatever switch could exist for those who want it as a global "no internet please" switch.

**Comment by akramer-zibra** [comment]
Guess an initial configuration with less complexity in its network range could also be benefictial for experienced users. Lets imagine predefined configuration profiles for common setups (local network only, etc.). Such profiles could be a valueable share of experience and a good starting point for experts to tweak their custom setup.

**Comment by wj7998** [comment]
yes, fully agree on "local only" mode make things much easier...

**Comment by erxbout** [comment]
+1 would also think that this would be useful

**Comment by Kaya-Sem** [comment]
I am willing to put some time into this, will hear from me again

**Comment by pycaw** [comment]
Well, LAN-only mode was described as "niche use case" https://github.com/syncthing/syncthing/pull/10226#issuecomment-3146455621 by @imsodin but I think people here would argue otherwise. Even if @Kaya-Sem 's implementation wasn't the ideal one and didn't have pristinely clear title, this feature is definitely desired by many! Here I am struggling with abysmal transfer speeds because instead of my PC and handheld device communicating via LAN they do so with packets traversing half the world with 50 layers of encryption applied on top (of my movie). Makes no sense whatsoerver. And even with disabling global discovery on my device its non-LAN IP is used.

**Comment by tomasz1986** [comment]
> And even with disabling global discovery on my device its non-LAN IP is used.

Please open a new topic on the forum if that is the case, because by default, Syncthing does prioritise local connections, and only falls back to using relays, etc. when it cannot connect locally.

In addition, you can already limit Syncthing to LAN only. The issue here is about adding a switch that would make it a one command affair, without having to tweak the configuration manually. Please keep in mind though that that wouldn't help if your devices cannot connect locally in the first place.

**Comment by pycaw** [comment]
> > And even with disabling global discovery on my device its non-LAN IP is used.
> 
> Please open a new topic on the forum if that is the case, because by default, Syncthing does prioritise local connections, and only falls back to using relays, etc. when it cannot connect locally.
> 
> In addition, you can already limit Syncthing to LAN only. The issue here is about adding a switch that would make it a one command affair, without having to tweak the configuration manually. Please keep in mind though that that wouldn't help if your devices cannot connect locally in the first place.

I wanted to look up how to restrict Syncthing (linux client) to LAN connections, couldn't determine how to do it by tweaking the options. Instead, I have come across all kinds of discussions but no docs about the issue. So, could you please point me to the place were this is described and what troubleshooting one might follow when what should happen doesn't? Lack of an intuitive interface to set this is still definitely not a niche a feature / use case! I think many don't realize that WAN is used instead of LAN and that's why their transfer speeds are slow. If they enabled something like LAN-only mode they could be notified accordingly.

**Comment by tomasz1986** [comment]
> I think many don't realize that WAN is used instead of LAN and that's why their transfer speeds are slow. If they enabled something like LAN-only mode they could be notified accordingly.

What I'm trying to say is that if a local connection hasn't been established by Syncthing automatically, there is no magic switch that will suddenly allow you to establish it just by tweaking some settings 😇. Usually, this is because there are firewalls or other restrictions on the network that block your devices from connecting directly to one another.

There is also a question what you mean by a "LAN only" mode. Is it just about restricting devices to local connections only? Or, do you want to block Syncthing from connecting to the outside world completely (which includes automatic upgrades, sending crash reports, etc.)?

You're right though that the documentation is lacking 🙁. I guess no-one has been bothered enough to write a page that would gather all the different settings in one place. I'm sure a properly submitted pull request regarding this matter would be welcome.

**Comment by pycaw** [comment]
> > I think many don't realize that WAN is used instead of LAN and that's why their transfer speeds are slow. If they enabled something like LAN-only mode they could be notified accordingly.
> 
> What I'm trying to say is that if a local connection hasn't been established by Syncthing automatically, there is no magic switch that will suddenly allow you to establish it just by tweaking some settings 😇. Usually, this is because there are firewalls or other restrictions on the network that block your devices from connecting directly to one another.
> 
> There is also a question what you mean by a "LAN only" mode. Is it just about restricting devices to local connections only? Or, do you want to block Syncthing from connecting to the outside world completely (which includes automatic upgrades, sending crash reports, etc.)?
> 
> You're right though that the documentation is lacking 🙁. I guess no-one has been bothered enough to write a page that would gather all the different settings in one place. I'm sure a properly submitted pull request regarding this matter would be welcome.

But the fact that I get no feedback as to why something is not happening that is supposed to happen is not good. My firewall is set up correctly, not entirely sure about the other side, Android though. I personally don't really care about how exactly LAN-only would work, what I firstly care about is that transfer happens fast (via LAN) and if that cannot happen I get the most precise error message possible. Secondly, it would indeed be great if I could set up related behavior on a neat interface. I also don't understand how WAN works if I disable global discovery on my Android, by the way. So I am still lost as to how I could set up "LAN-only mode" or disable transfer via WAN with what is offered now.
In my opinion, it is more and more important in this day and age to be able to restrict things from accessing the web if they don't need to. And if I need to stay 2 more minutes in my house so that a transfer finishes, that is a blatantly reasonable sacrifice to this end, as per my preferences.

**Comment by tomasz1986** [comment]
> But the fact that I get no feedback as to why something is not happening that is supposed to happen is not good.

As asked above, please open a new topic on the forum if you need help with your particular configuration. GitHub is used for clear issues and feature requests, but when it comes to individual support or other types of discussion, we use the forum for that.

**Comment by pycaw** [comment]
> > But the fact that I get no feedback as to why something is not happening that is supposed to happen is not good.
> 
> As asked above, please open a new topic on the forum if you need help with your particular configuration. GitHub is used for clear issues and feature requests, but when it comes to individual support or other types of discussion, we use the forum for that.

At this point I only wish to find information on configuring Syncthing for LAN-only mode which is said to be very much possible as of now. A mere link or set of links would be perfect -- and would benefit others too who land here wanting to do the very same thing as me.

**Comment by tomasz1986** [comment]
At the moment, there is no page in the Docs dealing specifically with that. If you want a simple way to limit your device connections to LAN only, you could set https://docs.syncthing.net/advanced/device-allowednetworks to something like `10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16`. This will block device connections that don't come from local IP addresses.

Please keep in mind that the one setting will only restrict connections between devices, and it won't affect any other connections that Syncthing makes to the Internet, e.g. discovery, relays listener, etc. This may also not work in corporate environments that use non-standard IP ranges for their network.

If you want to restrict Syncthing from connecting to the Internet completely, one of the forum topics listed in the first post in this issue (i.e. https://forum.syncthing.net/t/limit-syncthing-to-only-use-local-network-review-and-suggestions-appreciated/20117) deals with that quite comprehensively.

**Comment by pycaw** [comment]
@tomasz1986  Thank you for the help and the links. The per-device `allowedNetworks` setting is pretty much what I was looking for.

**Comment by lni-dev** [comment]
+1, I would love to see this as a feature too.
